### PR TITLE
Add support for multiline logging

### PR DESCRIPTION
**Pull request by gitlawr**
Address issue: https://github.com/rancher/rancher/issues/21900

Add a filter block for the concat plugin, do the validation accordingly.
Update `escapeString()` to address backslash issue.

### CODE REVIEW

File: pkg/controllers/user/logging/generator/validator.go [hunk 1]
```diff
@@ -124,16 +127,5 @@ func filterRubyCode(s string) error {
...
-		"\r", `\\r`,
-		"\f", `\\f`,
-		"\b", `\\b`,
-		"\"", `\\\"`,
-		"\\", `\\\\`,
-	)
-
-	escapeString := escapeReplacer.Replace(postDoc)
-	return fmt.Sprintf(`"%s"`, escapeString)
+	return fmt.Sprintf("%q", postDoc)
```
Comment (aiwantaozi) [inline pkg/controllers/user/logging/generator/validator.go]: This convert is the simulation from Fluentd. Base on the doc https://docs.fluentd.org/configuration/config-file#in-double-quoted-string-literal-is-escape-character and the test case https://github.com/fluent/fluentd/blob/v1.6.3/test/config/test_literal_parser.rb

For example, original user input is `firstline\nsecondline`, and we parse to `firstline\\nsecondline`, fluentd will parse back to `firstline\nsecondline`

File: pkg/controllers/user/logging/generator/templatefilter.go [hunk 2]
```diff
@@ -36,6 +36,25 @@ var FilterTemplate = `
 </filter>
 {{end}}
 
+{{define "filter-concat"}}
+{{- if .EnableMultiLineFilter}}
+<filter {{ .ContainerLogSourceTag }}.**>
+  @type concat
+  key log
+  stream_identity_key tag
```
[the remark this answers is not in the document]
Comment (gitlawr) [reply pkg/controllers/user/logging/generator/templatefilter.go]: concat multiline logs using the same tag, i.e., aggregate those from the same container

File: pkg/controllers/user/logging/generator/validator.go (hunk 1 above)
Comment (gitlawr) [reply pkg/controllers/user/logging/generator/validator.go]: https://github.com/rancher/rancher/blob/85dedb19bbdfabc5e81a8258b868cc1ca8261b0d/pkg/controllers/user/logging/generator/validator.go#L128
Take the above as an example.
Please note that the backslash inside backquote in Golang is not the escape character. so `\\t` is literally two backslashes and a `t`. When it is quoted as `"\\t"` in following codes, it becomes an escaped backslash and a `t` from the fluentd parser's view. And it does not match the original "\t" that stands for a single horizontal tab.

**Comment by aiwantaozi** [comment]
need rebase.

**Pull request closed without merging**